tools/mo/openvino/tools/mo/front/common/partial_infer/utils.py

```diff
@@ -179,12 +179,19 @@ def int64_array(value: Union[Iterable[Union[float, int]], float, int]) -> np.nda
     return np.array(value, dtype=np.int64)
 
 
+def float32_array(value: Union[Iterable[Union[float, int]], float, int]) -> np.ndarray:
+    return np.array(value, dtype=np.float32)
+
+
 def float_array(value: Union[Iterable[Union[float, int]], float, int]) -> np.ndarray:
-    return np.array(value, dtype=np.float64)
+    return float32_array(value)
 
 
-def float32_array(value: Union[Iterable[Union[float, int]], float, int]) -> np.ndarray:
-    return np.array(value, dtype=np.float32)
+def mo_array(value: Union[Iterable[Union[float, int]], float, int], dtype=None) -> np.ndarray:
+    x = np.array(value, dtype=dtype)
+    if not isinstance(value, np.ndarray) and x.dtype == np.float64 and dtype != np.float64:
+        x = x.astype(np.float32)
```

Comment:
what if in some code we have a list `x = [1,2,3]` and I call `np.array(x)`. So the result will be np.array of np.int32
In your case, if we substitute this with `mo_array(x)`, the result will be np.array of np.float32.

This function looks dangerous to me

Reply:
if np.array(x) return int32 mo_array will return int32 too